### ISSUE TITLE
cmd/vet: add s390x support

### ISSUE THREAD

**Issue by josharian** [golang/go#16373]
cmd/vet has support for checking assembly. It requires a few details about the architecture. See asmdecl.go. It'd be good to add s390x to it. Unless s390x is very unusual, it should be an easy two-line addition to the block of asmArch* vars.

/cc @billotosyr @mundaym

**Comment by mundaym** [comment]
This is a duplicate of #15454. The fix is waiting on a test.

**Comment by minux** [comment]
Merged into #15454.
